astroML/density_estimation/density_estimation.py

```diff
@@ -95,6 +95,9 @@ def eval(self, X):
         k = float(self.n_neighbors)
         ndim = X.shape[1]
 
+        # I hope assert is good here given that we only recognize the two methods
+        # and this should have been an error long time ago in all other cases.
+        assert self.method in ["simple", "bayesian"]
```

Comment:
No need for the comment in the code. assert can be fine, but as a rule of thumb they are not encouraged in code that is not a test, and the if/else below does the same anyway with a more useful for the users error message.

Reply:
ah, duh, funny the kind of things I miss sometimes.